streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -99,11 +99,11 @@ void createTasks(final Collection<TopicPartition> assignment) {
             throw new IllegalStateException(logPrefix + "consumer has not been initialized while adding stream tasks. This should not happen.");
         }
 
-        changelogReader.reset();
```

Comment:
We should not reset changelogReader since we still need its restored offset when closing the restoring tasks.